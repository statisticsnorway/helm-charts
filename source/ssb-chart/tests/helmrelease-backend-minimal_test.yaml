suite: test helmrelease for backend with minimal config

tests:
  - it: should render Deployment
    templates:
      - deployment.yaml
    values:
      - ./values/helmrelease-minimal.yaml
    set:
      appType: "backend"
    release:
      name: unittest-app
      namespace: unittest
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Deployment
      - matchSnapshot: {}

  - it: should render NetworkPolicy
    templates:
      - network-policy.yaml
    values:
      - ./values/helmrelease-minimal.yaml
    set:
      appType: "backend"
    release:
      name: unittest-app
      namespace: unittest
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: NetworkPolicy
      - matchSnapshot: {}

  - it: should render ServiceAccount
    templates:
      - serviceaccount.yaml
    values:
      - ./values/helmrelease-minimal.yaml
    set:
      appType: "backend"
    release:
      name: unittest-app
      namespace: unittest
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ServiceAccount
      - matchSnapshot: {}

  - it: should render Service
    templates:
      - service.yaml
    values:
      - ./values/helmrelease-minimal.yaml
    set:
      appType: "backend"
    release:
      name: unittest-app
      namespace: unittest
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Service
      - matchSnapshot: {}

  - it: should render RequestAuthentication
    templates:
      - istio-authentication.yaml
    values:
      - ./values/helmrelease-minimal.yaml
    set:
      appType: "backend"
    release:
      name: unittest-app
      namespace: unittest
    documentIndex: 0
    asserts:
      - hasDocuments:
          count: 2
      - isKind:
          of: RequestAuthentication
      - matchSnapshot: {}

  - it: should render AuthorizationPolicy (require JWT)
    templates:
      - istio-authentication.yaml
    values:
      - ./values/helmrelease-minimal.yaml
    set:
      appType: "backend"
    release:
      name: unittest-app
      namespace: unittest
    documentIndex: 1
    asserts:
      - hasDocuments:
          count: 2
      - isKind:
          of: AuthorizationPolicy
      - matchSnapshot: {}

  - it: should render AuthorizationPolicy
    templates:
      - authorization-policy.yaml
    values:
      - ./values/helmrelease-minimal.yaml
    set:
      appType: "backend"
    release:
      name: unittest-app
      namespace: unittest
    asserts:
      - hasDocuments:
          count: 2
      - isKind:
          of: AuthorizationPolicy
      - matchSnapshot: {}
