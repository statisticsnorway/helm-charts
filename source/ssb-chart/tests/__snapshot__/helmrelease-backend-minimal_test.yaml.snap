should render Deployment:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: unittest-app
        billing-project: ssb-unittest
        chart: ssb-chart-4.0.1
        heritage: Helm
        release: unittest-app
      name: unittest-app
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: unittest-app
          release: unittest-app
      template:
        metadata:
          labels:
            app: unittest-app
            release: unittest-app
        spec:
          containers:
            - image: ssb.no/unittest-app:1.0
              imagePullPolicy: Always
              name: unittest-app-cont
              ports:
                - containerPort: 80
                  name: http-main
              resources:
                limits:
                  memory: 128Mi
                requests:
                  cpu: 100m
                  memory: 128Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                privileged: false
                readOnlyRootFilesystem: true
                runAsGroup: 1000
                runAsNonRoot: true
                runAsUser: 1000
                seccompProfile:
                  type: RuntimeDefault
              volumeMounts:
                - mountPath: /tmp
                  name: tmp-volume
          securityContext:
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: unittest-app-sa
          volumes:
            - emptyDir: {}
              name: tmp-volume
should render Service:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: unittest-app
        billing-project: ssb-unittest
        chart: ssb-chart-4.0.1
        heritage: Helm
        release: unittest-app
      name: unittest-app
    spec:
      ports:
        - name: http-main
          port: 80
          targetPort: 80
      selector:
        app: unittest-app
        release: unittest-app
      type: ClusterIP
should render ServiceAccount:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app: unittest-app
        billing-project: ssb-unittest
        chart: ssb-chart-4.0.1
        heritage: Helm
        release: unittest-app
      name: unittest-app-sa
